packages/headless/src/controllers/core/context/headless-core-context.ts

```diff
@@ -95,6 +106,9 @@ export interface ContextState {
  * @param props - The configurable `Context` controller properties.
  *
  * @returns a `Context` controller instance.
+ *
+ * @group Core controllers
+ * @category Context
  */
 export function buildCoreContext(
```

Comment:
This one is weird as well it is only used in headless-context and in headless-strict-context-factory. @louis-bompart @y-lakhdar, I saw you two worked on those controllers. Can you give more context ?

It should not be in core/ but in search/

Reply:
That kind of thing happens in a lot of controllers, that's why I include a lot of "Core controllers". For a lot of Search components, the "buildX" method is exposed as a normal "Search controller", but the state and interface are in a different, /core file.
For example, `buildCategoryFacet` is in headless-category-facets.ts, 
<img width="389" alt="Screenshot 2024-10-11 at 3 16 46 PM" src="https://github.com/user-attachments/assets/a24936c4-6d67-486b-b140-d8decd4d1cd6">
but the state and interface are in headless-core-category-facets.ts
<img width="1332" alt="Screenshot 2024-10-11 at 3 19 00 PM" src="https://github.com/user-attachments/assets/bf1191ab-d885-451d-b856-07203923e97b">

Reply:
Furthermore, the core controllers states and interfaces can be referenced from builder functions accessible through different entry points. For instance, here the Insight CategoryFacet links to the core category facet interface and state
<img width="1318" alt="Screenshot 2024-10-11 at 3 24 30 PM" src="https://github.com/user-attachments/assets/b1604cb0-2523-4a4e-91c0-7647938e4915">